### DISCUSSION TITLE
Make collection literal IEnumerable<T> by default

### DISCUSSION THREAD

**Discussion by virzak** [dotnet/csharplang#7884] [Language Ideas]
When I want to create a FrozenSet from collection literal, it must be cast to `IEnumerable` first.

```cs
using System.Collections.Frozen;
using System.Collections.Generic;

var fs = ((IEnumerable<int>)[1]).ToFrozenSet();
```

Should it implement IEnumerable<T> by default?

**Comment by yaakov-h** [comment]
"must" is a strong word, you could also do:

```cs
using System.Collections.Frozen;
var fs = FrozenSet.ToFrozenSet([1]);
```

**Comment by ufcpp** [comment]
https://github.com/dotnet/csharplang/blob/main/proposals/collection-expressions-next.md#natural-type
https://github.com/dotnet/csharplang/blob/main/meetings/working-groups/collection-literals/CL-2023-06-26.md#extension-method-lookup
